overlore/db/handler.py

```diff
@@ -195,39 +188,46 @@ def init(self, path: str = "./events.db") -> DatabaseHandler:
         self.realms = Realms.instance().init()
         return self
 
-    def get_by_id(self, event_id: int) -> Any:
+    def get_by_id(self, event_id: int) -> StoredEvent:
+        # lock mutex
         self.__lock()
         cursor = self.db.cursor()
-        cursor.execute("SELECT id, type, importance, ts, metadata FROM events WHERE id=?", (event_id,))
-        ret = cursor.fetchall()
-        events_pos = self.get_events_pos_by_id(event_id)
-        ret = list(ret) + events_pos
+        cursor.execute(
+            "SELECT id, type, importance, ts, metadata, X(active_pos), Y(active_pos), X(passive_pos), Y(passive_pos)"
+            " FROM events WHERE id=?",
+            (event_id,),
+        )
+        record = cursor.fetchall()
+        # unlock mutex
         self.__release()
-        return ret
+        record = list(record[0])
+        record = record[:5] + [(record[5], record[6]), (record[7], record[8])]
+        return record
 
-    def get_events_pos_by_id(self, event_id: int) -> Any:
-        query = """SELECT X(active_pos), Y(active_pos), X(passive_pos), Y(passive_pos) from events WHERE id=?"""
-        cursor = self.db.cursor()
-        cursor.execute(query, (event_id,))
-        records = cursor.fetchall()
-        return [(records[0][0], records[0][1]), (records[0][2], records[0][3])]
-
-    def get_all(self) -> Any:
-        query = """SELECT * from events"""
+    def get_all(self) -> list[StoredEvent]:
+        # lock mutex
+        self.__lock()
+        query = """SELECT id, type, importance, ts, metadata, X(active_pos), Y(active_pos), X(passive_pos), Y(passive_pos) from events"""
         cursor = self.db.cursor()
         cursor.execute(query)
         records = cursor.fetchall()
-        print("Total rows are:  ", len(records))
-        print("Printing each row")
-        for row in records:
-            for elem in row:
-                print(elem)
-            print("\n")
+        # unlock mutex
+        self.__release()
+        records = [[*list(tup[:5]), (tup[5], tup[6]), (tup[7], tup[8])] for tup in records]
         return records
 
-    def process_event(self, event: dict[str, Any]) -> int:
-        if self.__get_event_type(event) == EventKeyHash.COMBAT_OUTCOME.value:
-            parsed_event = self.__parse_combat_outcome_event(event)
+    def process_event(self, event: ToriiEvent) -> int:
+        event_emitted = event.get("eventEmitted")
+        if not event_emitted:
+            raise RuntimeError("eventEmitted no present in event")
```

Comment:
hmm, thinking about whether we want the whole program to crash when events arent in the shape we expect. i think in the stage we are at now, this makes sense, but potentially in the future we would be good just logging these errors and keeping the system running